lib/dry/system/auto_registration_config.rb

```diff
@@ -0,0 +1,22 @@
+module Dry
+  module System
+    # @api private
+    AutoRegistraionConfig = Struct.new(:instance, :exclude) do
```

Comment:
Perhaps we could rename this to `Dry::System::AutoRegistrar::Configuration`? We typically spell the `configuration` word out in full for classes like this. And by namespacing it, we make it clear that the configuration relates to the `AutoRegistrar` only, and it means we can just refer to its  constant as `Configuration` from within the `AutoRegistrar` class.